### ISSUE TITLE
XWayland Display Freezes When Moving Window to Workspace and Switching to Workspace in Single "swaymsg" invocation

### ISSUE THREAD

**Issue by spiralman** [swaywm/sway#6168]
- **Sway Version:**
  - sway version v1.5

- **Debug Log:**
  - ~I will try to get a debug log ASAP, but wanted to capture the debugging I had already done. Here's the non-debug log:~

Debug log here: https://gist.github.com/spiralman/94e7e78f8819930914f387d01a108f71

Below is the non-debug log I originally captured, for posterity:

<details>
  <summary>Click to Expand Logs</summary>
(note the `xcb error: op 18:0, code3` is logged even when nothing goes wrong; see below for my workaround).

```
Apr 08 17:42:24 localhost.localdomain /usr/libexec/gdm-wayland-session[88092]: 00:15:56.908 [ERROR] [xwayland/xwm.c:1286] xcb error: op 18:0, code 3, sequence 1805, value 29360132
Apr 08 17:42:24 localhost.localdomain /usr/libexec/gdm-wayland-session[98758]: [
Apr 08 17:42:24 localhost.localdomain /usr/libexec/gdm-wayland-session[98758]:   {
Apr 08 17:42:24 localhost.localdomain /usr/libexec/gdm-wayland-session[98758]:     "success": true
Apr 08 17:42:24 localhost.localdomain /usr/libexec/gdm-wayland-session[98758]:   },
Apr 08 17:42:24 localhost.localdomain /usr/libexec/gdm-wayland-session[98758]:   {
Apr 08 17:42:24 localhost.localdomain /usr/libexec/gdm-wayland-session[98758]:     "success": true
Apr 08 17:42:24 localhost.localdomain /usr/libexec/gdm-wayland-session[98758]:   }
Apr 08 17:42:24 localhost.localdomain /usr/libexec/gdm-wayland-session[98758]: ]
Apr 08 17:42:26 localhost.localdomain /usr/libexec/gdm-wayland-session[91834]: [04/08/21, 17:42:26:010] info: Store: UPDATE_SETTINGS {
Apr 08 17:42:26 localhost.localdomain /usr/libexec/gdm-wayland-session[91834]:   "mainWindowSettings": {
Apr 08 17:42:26 localhost.localdomain /usr/libexec/gdm-wayland-session[91834]:     "fullScreen": false,
Apr 08 17:42:26 localhost.localdomain /usr/libexec/gdm-wayland-session[91834]:     "maximized": true,
Apr 08 17:42:26 localhost.localdomain /usr/libexec/gdm-wayland-session[91834]:     "bounds": {
Apr 08 17:42:26 localhost.localdomain /usr/libexec/gdm-wayland-session[91834]:       "width": 1024,
Apr 08 17:42:26 localhost.localdomain /usr/libexec/gdm-wayland-session[91834]:       "height": 768,
Apr 08 17:42:26 localhost.localdomain /usr/libexec/gdm-wayland-session[91834]:       "x": 448,
Apr 08 17:42:26 localhost.localdomain /usr/libexec/gdm-wayland-session[91834]:       "y": 156
Apr 08 17:42:26 localhost.localdomain /usr/libexec/gdm-wayland-session[91834]:     }
Apr 08 17:42:26 localhost.localdomain /usr/libexec/gdm-wayland-session[91834]:   }
Apr 08 17:42:26 localhost.localdomain /usr/libexec/gdm-wayland-session[91834]: }
Apr 08 17:42:26 localhost.localdomain /usr/libexec/gdm-wayland-session[91834]: [04/08/21, 17:42:26:019] info: [DESKTOP-SIDE-EFFECT] Update from desktop for keys  ["settings"]
```
</details>

- **Configuration File:**

```
bindsym $mod+Shift+equal exec 'swaymsg -t get_workspaces | jq -r ".[].name" | dmenu | xargs -I % swaymsg move container to workspace "%", workspace "%"'
```

- **Description:**
  - Given the above key binding, if I make an XWayland window active (tested with Chrome), then execute the keybinding
  - The entire XWayland session freezes and all XWayland apps stop redrawing correctly
    - If the window is resized, the contents aren't redrawn to reflect the new size
    - The window appears unresponsive to user input
    - Sometimes the UI appears to "flicker"
    - The mouse cursor changes based on where elements should be in the newly sized window, but clicking appears to have no effect. (i.e. if there was a link on the right side of the screen in Chrome, the cursor changes to the "pointer" when I move the mouse there).
    - The app appears to be still running (if I have Google Meet running, I can still hear the person I'm talking to, and video from my camera is still updating on the screen, but in the wrong place).
    - Xwayland is also still running in this state. If I kill it, it kills all the X11 apps, but then everything comes back up correctly when I restart them.
  - If I use two separate invocations of `swaymsg` to move the active window to a workspace and *then* select the workspace, it works correctly:

```
bindsym $mod+Shift+equal exec 'export WORKSPACE=$(swaymsg -t get_workspaces | jq -r ".[].name" | dmenu) && swaymsg move container to workspace "$WORKSPACE" && swaymsg workspace "$WORKSPACE"'
```
In this case, the logs do not contain the `info: Store: UPDATE_SETTINGS`, but all the other messages are the same, other than the JSON formatting of 2 `swaymsg` outputs instead of 1.

I can try to capture a debug log later, but probably not until the weekend. If there is anything else useful to capture, please let me know.

**Comment by kennylevinsen** [comment]
Please test sway 1.6.

**Comment by spiralman** [comment]
I installed sway 1.6 in a "toolbox" running Fedora 34 beta (there were too many outdated build deps to easily get it working in Fedora 33) and wasn't able to reproduce the issue. Note, though, that the setup was somewhat different, due to the toolbox (i.e. I was just using xeyes instead of chrome, etc).

I think that the issue is fixed. Once Fedora 34 is released, later this month, I can test again outside the podman/toolbox environment, just to confirm.

Debug log trying to reproduce the issue in 1.6, if it's important: https://gist.github.com/spiralman/7c244ddf0283554ab1054a7eb54eb8ad

**Comment by spiralman** [comment]
I tested fully after upgrading to Sway 1.6 in Fedora 34, and cannot reproduce this issue, so I believe it is fixed. Thanks!
